pkg/client/client.go

```diff
@@ -205,6 +208,63 @@ func NewClientFromKeys(accessKey, secretAccessKey, region string) (Client, error
 	}, nil
 }
 
+type describeRegionsData struct {
+	err                   error
+	describeRegionsOutput *ec2.DescribeRegionsOutput
+	lastUpdated           time.Time
+}
```

Comment:
Any data we are storing like this needs to have a mutex on it. Perhaps instead we can create a cache structure that is owned by the reconciler? That way it is a property of something rather than a global variable. A `RegionCache` would become an argument to validated client

Reply:
I used global variable here because it reduced the number of places that required changes. I've made a separate commit that moves the cache ownership to the actuator. We can merge both or just the first one. Depending on what you like more.